packages/mui-material/src/PaginationItem/PaginationItem.js

```diff
@@ -244,6 +244,12 @@ const PaginationItem = React.forwardRef(function PaginationItem(inProps, ref) {
     className,
     color = 'standard',
     component,
+    components = {
```

Comment:
@michaldudak I suggested here that we use the `components` API, but seeing the code, these are used depending on the `type`. I still think it's better to use the `components` API than adding four new props, but let me know if you disagree

Reply:
I think it's a good approach. The pattern is similar to the unstyled Input - it accepts the `components.Textarea` but uses it only conditionally.